cli/command/secret/create.go

```diff
@@ -43,9 +46,29 @@ func runSecretCreate(dockerCli *command.DockerCli, options createOptions) error
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	secretData, err := ioutil.ReadAll(os.Stdin)
+	if options.file == "" {
+		return fmt.Errorf("Please specify either a file name or STDIN ('-')  with --file")
+	}
+
+	var in io.Reader
+	if options.file == "-" {
+		// To avoid getting stuck, verify that a file is given in STDIN
+		if dockerCli.In().IsTerminal() {
+			return fmt.Errorf("Requested from STDIN ('-'), but STDIN is empty")
```

Comment:
I think it should be okay to type in the secret on stdin. I don't know of any other unix tools that disallow this. And the error message doesn't seem right for this case.